101-key-vault-create/azuredeploy.json

```diff
@@ -5,118 +5,135 @@
+        "description": "Location for the key vault."
       }
     },
-    "objectId": {
+    "tenantId": {
       "type": "string",
+      "defaultValue": "[subscription().tenantId]",
       "metadata": {
-        "description": "Object Id of the AD user. Get using Get-AzureRmADUser or Get-AzureRmADServicePrincipal cmdlets"
-      }
-    },
-    "keysPermissions": {
-      "type": "array",
-      "defaultValue": [
-        "list"
-      ],
-      "metadata": {
-        "description": "Permissions to keys in the vault. Valid values are: all, create, import, update, get, list, delete, backup, restore, encrypt, decrypt, wrapkey, unwrapkey, sign, and verify."
-      }
-    },
-    "secretsPermissions": {
-      "type": "array",
-      "defaultValue": [
-        "list"
-      ],
-      "metadata": {
-        "description": "Permissions to secrets in the vault. Valid values are: all, get, set, list, and delete."
+        "description": "Tenant Id of the Azure subscription used for authenticating requests to the key vault. Get it by using Get-AzSubscription cmdlet."
       }
     },
-    "skuName": {
+    "objectId": {
       "type": "string",
-      "defaultValue": "Standard",
-      "allowedValues": [
-        "Standard",
-        "Premium"
-      ],
-      "metadata": {
-        "description": "SKU for the vault"
-      }
-    },
-    "enableVaultForDeployment": {
-      "type": "bool",
-      "defaultValue": false,
-      "allowedValues": [
-        true,
-        false
-      ],
       "metadata": {
-        "description": "Specifies if the vault is enabled for a VM deployment"
+        "description": "The object ID of the Azure AD user for the key vault. Get it by using Get-AzADUser or Get-AzADServicePrincipal cmdlets."
       }
     },
-    "enableVaultForDiskEncryption": {
-      "type": "bool",
-      "defaultValue": false,
-      "allowedValues": [
-        true,
-        false
-      ],
+    "secretName": {
+      "type": "string",
+      "defaultValue": "vmAdminPassword",
       "metadata": {
-        "description": "Specifies if the azure platform has access to the vault for enabling disk encryption scenarios."
+        "description": "Name of the secret that you want to create."
       }
     },
-    "enabledForTemplateDeployment": {
-      "type": "bool",
-      "defaultValue": false,
-      "allowedValues": [
-        true,
-        false
-      ],
+    "secretValue": {
+      "type": "securestring",
       "metadata": {
-        "description": "Specifies whether Azure Resource Manager is permitted to retrieve secrets from the key vault."
-      }
-    },
-    "location": {
-      "type": "string",
-      "defaultValue": "[resourceGroup().location]",
-      "metadata": {
-        "description": "Location for all resources."
+        "description": "Value of the secret that you want to create."
       }
     }
   },
+  "variables": {
+    "enabledForDeployment": false,
```

Comment:
revert this change, these should be parameters

Reply:
Done.  I also move the other two enabled values to the parameter section.